tee-worker/identity/enclave-runtime/src/rpc/common_api.rs

```diff
@@ -506,6 +516,105 @@ pub fn add_common_api<Author, GetterExecutor, AccessShieldingKey, OcallApi, Stat
 			Err(_) => Ok(json!(compute_hex_encoded_return_error("Could not parse params"))),
 		}
 	});
+
+	let local_state = state.clone();
+
+	io_handler.add_sync_method("omni_UploadIDGraph", move |params: Params| {
+		debug!("worker_api_direct rpc was called: omni_UploadIDGraph");
```

Comment:
I would suggest putting this in startup code, guarded by CLI argument.

Currently anybody can call this method - it will flood parachain.
Another thing is: upload should be triggered **after** sidechain's imp pallet becomes immutable and **before** parachain storage becomes mutable - otherwise there will be conflicts/outdated account stores.

Reply:
I didn't go for this as I wanted to save the communication between non-enclave and enclave,  as it's a one-time thing. `omni_UploadIDGraph` could be called multiple times - e.g. right before we switch to the new on-chain IDGraph, the whole process should be done within 1 minute - like it's hard for any abusers to predict when we are going to do the upgrade.

Let me change it to start-up sub-command. In that way we'll stop worker, do the upload (so that nothing changes at the same time), and restart it with newer version that supports on-chain Graph

Reply:
`start-up sub-command` that would be a better solution. Completely offline. No worries for any potential hacks.

Reply:
I updated it